packages/not-next/src/hooks/useClientWithStateManager.ts

```diff
@@ -1,48 +1,44 @@
+import { useShopId } from "./useShopId.ts";
+import { useQuery } from "./useQuery.ts";
+import { useKeycard } from "./useKeycard.ts";
+import { useRelayEndpoint } from "./useRelayEndpoint.ts";
+import { ClientWithStateManager } from "../ClientWithStateManager.ts";
 
 export function useClientWithStateManager() {
-  // const { clientStateManager, setClientStateManager } = useContext(
-  //   MassMarketContext,
-  // );
-  // const { isMerchantKeyCard, isGuestKeyCard, keycard } = useKeycard();
-  // const shopId = useShopId();
-  // const shopPublicClient = usePublicClientForRegistry();
-  // const relayEndpoint = useRelayEndpoint();
-  // //Create a new ClientWithStateManager instance when shopId changes
-  // useEffect(() => {
-  //   if (
-  //     shopId &&
-  //     relayEndpoint &&
-  //     shopPublicClient &&
-  //     clientStateManager.shopId !== shopId // make sure we are not creating a new instance with the same shopId
-  //   ) {
-  //     const csm = new ClientWithStateManager(
-  //       shopPublicClient,
-  //       shopId,
-  //       relayEndpoint,
-  //     );
-  //     setClientStateManager(csm);
-  //   }
-  // }, [shopId]);
-  // const result = useQuery(async () => {
-  //   if (isMerchantKeyCard) {
-  //     await clientStateManager.setClientAndConnect(keycard);
-  //     await clientStateManager.sendMerchantSubscriptionRequest();
-  //   } else if (!keycard) {
-  //     //If no keycards are cached, create relayClient with guest wallet, then connect without enrolling a kc or authenticating.
-  //     await clientStateManager.sendGuestSubscriptionRequest();
-  //   } else if (isGuestKeyCard) {
-  //     //If guestCheckout keycard is cached, connect, authenticate, and subscribe to orders.
-  //     await clientStateManager.setClientAndConnect(keycard);
-  //     await clientStateManager.sendGuestCheckoutSubscriptionRequest();
-  //   }
-  // }, [keycard, clientStateManager]);
-  // return { clientStateManager, ...result };
+  const { clientStateManager, setClientStateManager } = useContext(
+    MassMarketContext,
+  );
+  const [keycard] = useKeycard();
+  const { relayEndpoint } = useRelayEndpoint();
+  const { shopId } = useShopId();
+  const { shopPublicClient } = usePublicClient();
+
+  const currentClientStateManager = useMemo(() => {
+    if (
+      shopId &&
+      relayEndpoint &&
+      shopPublicClient &&
+      clientStateManager?.shopId !== shopId
+    ) {
+      return new ClientWithStateManager(
+        shopPublicClient,
+        shopId,
+        relayEndpoint,
+      );
+    }
+    return clientStateManager;
+  }, [shopId, relayEndpoint, shopPublicClient, clientStateManager?.shopId]);
+
+  useEffect(() => {
+    if (currentClientStateManager !== clientStateManager) {
+      setClientStateManager(currentClientStateManager);
+    }
+  }, [currentClientStateManager, clientStateManager]);
+
+  const result = useQuery(async () => {}, [keycard, clientStateManager]);
```

Comment:
leaving this body blank for now, since I have to rewrite the keycard subscription logic with the useKeycard hook